aapcs64/aapcs64.rst

```diff
@@ -3111,6 +3112,34 @@ instead.
 The SVE tuple types are mangled using their ``arm_sve.h`` names
 (``svBASExN_t``).
 
+Types which have an SME streaming or ZA interface should include an additional suffix as described in the table below:
```

Comment:
Adding suffixes like this would conflict with the existing mangling scheme for types.  E.g. `sm` would normally indicate an extra `short` argument followed by an extra `unsigned long` argument ([spec](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangling)):

```
$ c++filt
_Z1fPFu10__SVInt8_tsmE
f(__SVInt8_t (*)(short, unsigned long))
```

I wondered whether we could use a vendor-specific qualifier (`U…`), but function argument types are understandably mangled without a qualifier.  (For example, there is no difference between `f(const int)` and `f(int)`.)

For `__attribute__((arm_sve_vector_bits(N)))`, we used a fake template type `__SVE_VLS<…, N>` to attach the vector length N to the underlying standard SVE vector type.  Perhaps we could do the same sort of thing here: have a fake template type that wraps the unadorned function type and adds information about the streaming mode and shared state?  This would probably mean encoding that information as C++ expressions.

Reply:
Sorry, I hopelessly confused myself while writing and rewriting the comment above, so gave the wrong reason.  The problem isn't that function argument types have no qualifiers, but that function types themselves have only cv-qualifiers.  The spec hard-codes markup for transactional safety (`Tx`), but AFAIK there's no general mechanism for adding vendor extensions at the function type level.

The conclusion is the same though :-)